docs/README.md

```diff
@@ -1,47 +1,190 @@
-`ignore-check.kube-linter.io/privileged: "This deployment needs to run as privileged because it needs kernel access"`.
-
-To ignore _all_ checks for a specific object, you can use the special annotation key `kube-linter.io/ignore-all`.
+# KubeLinter
+
+KubeLinter analyzes Kubernetes YAML files and Helm charts and checks them
+against various best practices, with a focus on production readiness and
+security.
+
+KubeLinter runs sensible default checks designed to give you useful information
+about your Kubernetes YAML files and Helm charts. Use it to check early and
+often for security misconfigurations and DevOps best practices. Some common
+issues that KubeLinter identifies are running containers as a non-root user,
+enforcing least privilege, and storing sensitive information only in secrets.
+
+KubeLinter is configurable, so you can enable and disable checks and create your
+custom checks, depending on the policies you want to follow within your
+organization. When a lint check fails, KubeLinter also reports recommendations
+for resolving any potential issues and returns a non-zero exit code.
+
+> [!WARNING]
+> KubeLinter is at an early stage of development. There may be breaking changes
+> in the future to the command usage, flags, and configuration file formats.
+> However, we encourage you to use KubeLinter to test your environment YAML
+> files, see what breaks, and
+> [contribute](https://github.com/stackrox/kube-linter/blob/main/CONTRIBUTING.md)
+> to its development.
+
+## Installing KubeLinter
+
+### Using Go
+
+To install by using [Go](https://golang.org/), run the following command:
+
+```bash
+GO111MODULE=on go get golang.stackrox.io/kube-linter/cmd/kube-linter
+```
+Otherwise, download the latest binary from
+[Releases](https://github.com/stackrox/kube-linter/releases) and add it to your
+PATH.
+
+### Using Homebrew
+
+To install by using [Homebrew](https://brew.sh/) on macOS, Linux, and [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/about),
+run the following command:
+
+```bash
+brew install kube-linter
+```
+
+### Using Docker
+
+1. Get the latest KubeLinter Docker image:
+   ```bash
+   docker pull stackrox/kube-linter
+   ```
+1. Add current directory with your `yaml` file as a read only volume for the
+   `docker run` command:
+   ```bash
+   docker run -v ${PWD}:/tmp:ro --rm -i stackrox/kube-linter lint /tmp/pod.yaml
+   ```
+
+## Building from source
+
+> [!NOTE] Before you build, make sure that you have [installed Go](https://golang.org/doc/install).
+
+To build KubeLinter from source:
+
+1. Clone the KubeLinter repository:
+   ```bash
+   git clone git@github.com:stackrox/kube-linter.git
+   ```
+1. Compile the source code:
+   ```bash
+   make build
+   ```
+   This command compiles the source code and creates `kube-linter` binary files
+   for each platform in the `.gobin` folder.
+1. Verify that the compiled binary is working:
+   ```bash
+   .gobin/kube-linter version
+   ```
```

Comment:
We should tell people that they can optionally do one of the following to be able to invoke the binary as `kube-linter`.
 a) copy `.gobin/kube-linter` to some directory in their PATH or b) add `.gobin` to their PATH by running:
```bash
echo export PATH='"${PATH}:'"$(pwd)/.gobin"'"'
```
and pasting the output into their bashrc or other similar file.